Code/GraphMol/RGroupDecomposition/RGroupUtils.cpp

```diff
@@ -103,5 +105,53 @@ bool hasDummy(const RWMol &core) {
   }
   return false;
 }
+
+namespace {
+std::string MolToText(const ROMol &mol) {
+  bool hasQuery = false;
+  for (const auto atom : mol.atoms()) {
+    if (atom->hasQuery() && atom->getQuery()->getDescription() != "AtomNull") {
+      hasQuery = true;
+      break;
+    }
+  }
+  if (!hasQuery) {
+    for (const auto bond : mol.bonds()) {
+      if (bond->hasQuery()) {
+        hasQuery = true;
+        break;
+      }
+    }
+  }
+  if (!hasQuery) {
+    return MolToSmiles(mol);
+  } else {
+    return MolToSmarts(mol);
+  }
+}
 }  // namespace
 
+std::string toJSON(const RGroupRow &rgr, const std::string &prefix) {
```

Comment:
prefix isn't documented or used as far as I can tell, what is the purposes of this var?

Reply:
Will document it. It's there to make things more readable... it's a string which is added to each output line.
Currently it's being used to allow indentation